wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/interfaces/Statement.java

```diff
@@ -89,4 +89,11 @@
 	 * @return main value of the statement, or null
 	 */
 	Value getValue();
+	
+	/**
+	 * Returns the same statement, but with a different identifier.
+	 * This is useful when the existing identifier was empty and we need to 
+	 * attribute one before creating the statement in a remote Wikibase instance.
+	 */
+	Statement withStatementId(String id);
```

Comment:
It would be great to avoid this method and assign a statement id as soon a statement object is created. But I believe that in this case we would have troubles using the wbcreateclaim method that returns a new statementId for the newly created claim. There is maybe a nicer middle ground (but I don't see one yet).

Reply:
The problem with that is that you still need to be able to represent "statements that do not exist on the wikibase server yet"… If you give them an ID you will not be able to distinguish them from the existing ones. I think we rely on this distinction in StatementUpdate (but I am not entirely sure).

One other issue with assigning ids at creation is that it relies on the random number generator, so it might make things a bit more complicated to test (or you need to pass on a GuidGenerator in many classes…) but I guess that can be overcome.

Reply:
> I think we rely on this distinction in StatementUpdate (but I am not entirely sure)

The Wikibase API returns the statement serialization when created or modified. I believe that in StatementUpdate what would be clean to do is just deserialize the returned statements and replace the pushed statement by the new one. This way we would avoid this setter and also get the value
 normalizations routines executed by wikibase also pushed back in the Java object.

Reply:
no no, we actually need to set the ID (generated by us) *before* sending the statement to Wikibase. That's really weird, but that's how the API works. They expect clients to generate these ids.

Reply:
Thank you. I have forgotten that you had to create a statement id when you want to create a statement with qualifier and references in one edit. Ok, so it make sense to have this method.